client/components/domain-search-v2/register-domain-step/index.jsx

```diff
@@ -517,6 +517,16 @@ class RegisterDomainStep extends Component {
 			} ),
 			total,
 			onAddItem: ( domain_name ) => {
+				const subdomainPosition = this.state.subdomainSearchResults.findIndex(
+					( subdomain ) => subdomain.domain_name === domain_name
+				);
+
+				if ( subdomainPosition !== -1 ) {
+					const subdomain = this.state.subdomainSearchResults[ subdomainPosition ];
+
+					return this.onAddDomain( subdomain, subdomainPosition, false );
+				}
+
```

Comment:
Not related to this change, but can we somehow be aware of cases where a suggestion that has been clicked isn't found in the list below (i.e., `suggestionPosition === -1`)? Just so we can fix this potential bug.

Reply:
Added track event logic